django/db/models/query_utils.py

```diff
@@ -332,5 +335,9 @@ def resolve_expression(self, *args, **kwargs):
     def as_sql(self, compiler, connection):
         # Resolve the condition in Join.filtered_relation.
         query = compiler.query
-        where = query.build_filtered_relation_q(self.condition, reuse=set(self.path))
+        # Add other usable aliases in the query to the reuse set.
+        # Check for if it can be used is in Query.join
+        reusable_aliases = self.path
+        reusable_aliases += list(query.alias_map.keys())
```

Comment:
This change allows lookups to use relations not in the direct mapping of the FilteredRelation